src/ctap2/utils.rs

```diff
@@ -1,14 +1,39 @@
 use serde::de;
-use serde_cbor::error::Result;
 use serde_cbor::Deserializer;
+use std::io::Read;
 
-pub fn from_slice_stream<'a, T>(slice: &'a [u8]) -> Result<(&'a [u8], T)>
+pub fn serde_parse_err<E: de::Error>(s: &str) -> E {
+    E::custom(format!("Failed to parse {s}"))
+}
+
+pub fn from_slice_stream<'a, T, R: Read, E: de::Error>(data: &mut R) -> Result<T, E>
 where
     T: de::Deserialize<'a>,
 {
-    let mut deserializer = Deserializer::from_slice(slice);
-    let value = de::Deserialize::deserialize(&mut deserializer)?;
-    let rest = &slice[deserializer.byte_offset()..];
+    let mut deserializer = Deserializer::from_reader(data);
+    de::Deserialize::deserialize(&mut deserializer)
+        .map_err(|x| serde_parse_err(&format!("{}: {}", stringify!(T), &x.to_string())))
+}
+
+// Parsing routines
```

Comment:
There's a copy of the `byteorder` crate in mozilla-central already, so you can use that if you'd like. Up to you.

Reply:
Tempting, but since I wanted to remove dependencies here, I think I'll stick with those 2 hand-written functions.